.github/workflows/solutions_ci.yml

```diff
@@ -0,0 +1,43 @@
+name: CI for SAF templates
+on:
+  pull_request:
+
+jobs:
+  check-solution-templates-changes:
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    outputs:
+      changed: ${{ steps.check.outputs.has_changes}}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Check for changes
+        id: check
+        run: |
+          CHANGES=$(git diff --name-only  ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }} --  './src/ansys/templates/python/solution' './src/ansys/templates/python/osl_solution')
+          if [ -z "$CHANGES" ]; then
+            echo "has_changes=false" >> $GITHUB_OUTPUT
+            echo "No changes in the solution templates."
+          else
+            echo "has_changes=true" >> $GITHUB_OUTPUT
+            echo "Found changes in the solution templates."
+          fi
+
+  solutions-e2e-tests:
+    needs: [check-solution-templates-changes]
+    if: |
+      needs.check-solution-templates-changes.outputs.changed == 'true'
+    name: Trigger solutions e2e tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: github workflow dispatch
+        shell: bash
+        run: |
+          curl -L \
+          -X POST \
+          -H 'Accept: application/vnd.github+json' \
+          -H "Authorization: Bearer ${{ secrets.ANSYS_SOLUTIONS_E2E_TESTS }}"\
```

Comment:
To handle authentication with the GH CLI, you can do something like this: https://github.com/Solution-Applications/actions/pull/115/files#diff-cf5ec9da321459d0d1b98c7aeb3fb8e2836ff2a3da032b472c5f5e2720cb3223R84-R85

Reply:
Although note that since you are triggering a workflow, I don't think using the default `gh-token` will work, you will likely need to use `${{ secrets.ANSYS_SOLUTIONS_E2E_TESTS }}`